libs/ledgerjs/packages/cryptoassets/src/types.ts

```diff
@@ -1,119 +1,5 @@
-  name: string;
-  // the ticker name in exchanges / countervalue apis (e.g. BTC).
-  ticker: string;
-  // all units of a currency (e.g. Bitcoin have bitcoin, mBTC, bit, satoshi)
-  // by convention, [0] is the default and have "highest" magnitude
-  units: Unit[];
-  // a shorter version of code using the symbol of the currency. like Ƀ . not all cryptocurrencies have a symbol
-  symbol?: string;
-  // tells if countervalue need to be disabled (typically because colliding with other coins)
-  disableCountervalue?: boolean;
-  delisted?: boolean;
-  // some countervalue will have a ticker alias
-  countervalueTicker?: string;
-};
-
-/**
- *
- */
-export type TokenCurrency = CurrencyCommon & {
-  type: "TokenCurrency";
-  id: string;
-  ledgerSignature?: string;
-  contractAddress: string;
-  // the currency it belongs to. e.g. 'ethereum'
-  parentCurrency: CryptoCurrency;
-  // the type of token in the blockchain it belongs. e.g. 'erc20'
-  tokenType: string;
-  // indicates this is a compound token and it's "parent" erc20 have this id
-  compoundFor?: string;
-};
-
-/**
- *
- */
-export type FiatCurrency = CurrencyCommon & {
-  type: "FiatCurrency";
-};
-
-/**
- *
- */
-export type ExplorerView = {
-  tx?: string;
-  address?: string;
-  token?: string;
-};
-
-/**
- *
- */
-export type CryptoCurrency = CurrencyCommon & {
-  type: "CryptoCurrency";
-  // unique internal id of a crypto currency
-  id: string;
-  // define if a crypto is a fork from another coin. helps dealing with split/unsplit
-  forkedFrom?: string;
-  // name of the app as shown in the Manager
-  managerAppName: string;
-  // coin type according to slip44. THIS IS NOT GUARANTEED UNIQUE across currencies (e.g testnets,..)
-  coinType: number;
-  // the scheme name to use when formatting an URI (without the ':')
-  scheme: string;
-  // used for UI
-  color: string;
-  family: string;
-  blockAvgTime?: number;
-  // in seconds
-  supportsSegwit?: boolean;
-  supportsNativeSegwit?: boolean;
-  // if defined this coin is a testnet for another crypto (id)};
-  isTestnetFor?: string;
-  // TODO later we could express union of types with mandatory bitcoinLikeInfo for "bitcoin" family...
-  bitcoinLikeInfo?: {
-    P2PKH: number;
-    P2SH: number;
-    XPUBVersion?: number;
-    // FIXME optional as we miss some data to fill
-    hasTimestamp?: boolean;
-  };
-  ethereumLikeInfo?: {
-    chainId: number;
-    networkId?: number;
-    baseChain?: string;
-    hardfork?: string;
-  };
-  explorerViews: ExplorerView[];
-  terminated?: {
-    link: string;
-  };
-  deviceTicker?: string;
-};
-
-/**
- *
- */
-export type Currency = FiatCurrency | CryptoCurrency | TokenCurrency;
+// FIXME This should be the other way around, CryptoCurrencyIds should be defined on its own and cryptocurrenciesById should be derived from it.
+// That way CryptoCurrencyIds could be moved to the types-cryptoassets lib where it belongs.
+export type CryptoCurrencyIds = keyof typeof cryptocurrenciesById;
```

Comment:
The `currencies.ts` file in ledgerjs/cryptoassets is a bit sensitive, it's part of the CAL update process, it's documented in the dev portal... changing it has some impact for different people including external contributors, so I'd rather not make this change now.
The only impact of not doing it is that currencies are typed as `string` instead of `CryptoCurrencyIds`, not ideal but I figured it could be addressed later.

Reply:
This will be addressed there : https://ledgerhq.atlassian.net/browse/LIVE-1660